UnityProject/Assets/Scripts/Objects/Research/Portal.cs

```diff
@@ -102,19 +109,26 @@ public override bool WillAffectObject(GameObject eventData)
 
 		public override void OnObjectEnter(GameObject eventData)
 		{
-			Teleport(eventData);
+			if (isOnCooldown) return;
+			_ = Teleport(eventData);
 		}
 
-		private void Teleport(GameObject eventData)
+		private async Task Teleport(GameObject eventData)
```

Comment:
also I think puts it on a different thread so no accessing unity functions + put a try captain for logging errors because they won't Logged to the consul